vignettes/tables.Rmd

```diff
@@ -113,9 +113,145 @@ results$asDataFrame()
+
+# remove the column
+oldTable$removeColumn(test_column)
+newTable <- synStore(oldTable)
+
+# create a new Column
+newCol <- Column(name='IsTest', columnType='STRING', maximumSize=5)
+newCol <- synStore(newCol)
+
+# add the new column to the new table
+newTable$addColumn(newCol)
+newTable <- synStore(newTable)
+
+# copy the data over to the new column
+newQuery <- synTableQuery(sprintf("SELECT * FROM %s", newTable$properties$id))
+newData <- newQuery$asDataFrame(rowIdAndVersionInIndex=FALSE)
+newData['IsTest'] <- oldData['IsTest']
+
+# save the change
+write.csv(newData, newQuery$filepath, row.names=FALSE)
+synStore(Table(tableId, newQuery$filepath))
+
+# add the new data
+more_genes<-data.frame(
+    Name=c("not_sure"), 
+    Chromosome=c(2), 
+    Start=c(12345),
+    End=c(126000),
+    Strand=c('+'),
+    TranscriptionFactor=c(F),
+    IsTest=c('NA'))
+more_genes_file<-tempfile()
+write.csv(more_genes, file=more_genes_file, row.names=FALSE)
+synStore(Table(tableId, more_genes_file))
+```
+
+## Table Attached Files
+
+Synapse tables support a special column type called ‘File’ which contain a file handle, an identifier of a file stored in Synapse. Here’s an example of how to upload files into Synapse, associate them with a table and read them back later:
+
+```{r collapse=TRUE, eval=F}
+# example depends on synapseclient.uploadSynapseManagedFileHandle()
+# and not available until python client 1.7.3 is released
+new_cols <- list(
+    Column(name='artist', columnType='STRING', maximumSize=50),
+    Column(name='album', columnType='STRING', maximumSize=50),
+    Column(name='year', columnType='INTEGER'),
+    Column(name='catalog', columnType='STRING', maximumSize=50),
+    Column(name='cover', columnType='FILEHANDLEID'))
+new_schema <- synStore(Schema(name='Jazz Albums', columns=new_cols, parent=project))
+
+new_data <- data.frame(
+  artist = c("John Coltrane", "Sonny Rollins", "Sonny Rollins", "Kenny Burrel"),
+  album = c("Blue Train", "Vol. 2", "Newk's Time", "Kenny Burrel"),
+  year = c(1957, 1957, 1958, 1956),
+  catalog = c("BLP 1577", "BLP 1558", "BLP 4001", "BLP 1543")
+)
+
+# writing some temp files to upload or pointing to existing files in your system
+
+files <- c("coltraneBlueTrain.jpg", "rollinsBN1558.jpg", "rollinsBN4001.jpg", "burrellWarholBN1543.jpg")
+
+# upload to filehandle service
+files <- lapply(files, function (f) {
+  temp_file <- tempfile(f)
+  write(f, file=temp_file)
+  synUploadSynapseManagedFileHandle(f)
+  })
+# get the filehandle ids
+fileHandleIds <- sapply(files, function(f) f[1]$id)
+new_data["covers"] <- fileHandleIds
+
+temp_file<-tempfile()
+write.csv(new_data, file=temp_file, row.names=FALSE)
+synStore(Table(new_schema$properties$id, temp_file))
+```
+
+## Set Annotations
+A table is a Synapse entity. Annotations on table works the same way with annotations on any other entity types. Please visit [synapser vignettes](synapser.html) to read more about how to set annotations on an entity.
```

Comment:
A table is a Synapse entity. -> A table schema is a Synapse entity.

Add an example